src/utils.ts

```diff
@@ -112,24 +98,11 @@ export function transformTimeMeasuresToRelative(timeMeasures: TimeMeasure[]): Ti
         .sort((a, b) => a.timeMs - b.timeMs);
 }
 
-export function isValidUrl(input: string): boolean {
-    return URL_REGEX.test(input);
-}
-
-/**
- * Convert a string to a proper URL.
- */
-export function toProperUrl(url: string) {
-    // If it starts with a valid protocol, return as is
-    if (PROTOCOL_REGEX.test(url)) {
-        return url;
-    }
-
-    // If it matches a domain pattern, prepend http or https
-    if (DOMAIN_REGEX.test(url)) {
-        return `https://${url}`;
+export function safeUrl(input: string): URL | null {
```

Comment:
I'd call this function something like `interpretAsUrl`, return null if it's not http/https, and always return a string.